app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt

```diff
@@ -180,6 +187,20 @@ class ConversationMessagesViewModel @Inject constructor(
         }
     }
 
+    fun onResetSession(userId: UserId, clientId: String?) {
+        viewModelScope.launch {
+            conversationViewState =
+                when (withContext(dispatchers.io()) { resetSession(conversationId, userId, ClientId(clientId.orEmpty())) }) {
```

Comment:
I think we don't need to switch contexts here, the use case should handle that ( I updated that in kalium PR)